manage_breast_screening/notifications/management/commands/send_message_batch.py

```diff
@@ -0,0 +1,92 @@
+from datetime import datetime, timedelta
+from zoneinfo import ZoneInfo
+
+from django.core.management.base import BaseCommand, CommandError
+
+from manage_breast_screening.notifications.api_client import ApiClient
+from manage_breast_screening.notifications.models import (
+    Appointment,
+    Message,
+    MessageBatch,
+)
+
+TZ_INFO = ZoneInfo("Europe/London")
+
+
+class Command(BaseCommand):
+    """
+    Django Admin command which finds Appointment records which need batching up to send
+    to Communication Management API, and creates MessageBatch and Message records for them.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument("routing_plan_id")
+
+    def handle(self, *args, **options):
+        try:
+            routing_plan_id = options["routing_plan_id"]
+
+            self.stdout.write("Finding appointments to include in batch...")
+            appointments = Appointment.objects.filter(
+                starts_at__lte=self.schedule_date(), message__isnull=True
```

Comment:
question more for clarification - so we check whether to send a Message if an Appointment doesn't have a Message already? Just thinking out loud about edge cases, like if we wanted to mop up any failed messages we could look for Appointments that do have a Message but the status is `failed`? Should we hard code for now a OneToOne relationship between Appointments and Messages?